<?xml version="1.0"?>

<launch>
<arg name="pipeline" default="dirrt" />
<!--<arg name="mao_planner" default="ha_planner/DgacoPlannerManager" />-->
<arg name="mao_planner" default="pathplan/DIRRTStar" />
<arg name="load_param" default="load" />
<arg name="debug" default="false" />
<arg name="baseline" default="false"/>
<arg if="$(arg debug)" name="launch_prefix" default="xterm -e gdb -ex run  --args" />
<arg unless="$(arg debug)" name="launch_prefix" default="" />
<arg name="disable_collision" default="false" />

<rosparam if="$(eval pipeline == 'human_aware')" command="load" file="$(find hrc_simulator_moveit_config)/config/ha_planning.yaml"/>
<rosparam if="$(eval pipeline == 'dirrt')" command="load" file="$(find hrc_simulator_moveit_config)/config/dirrt_planning.yaml"/>
<!--<rosparam if="$(arg baseline)" command="load" file="$(find hrc_simulator_moveit_config)/config/dirrt_planning_base.yaml" ns="go_to_location_server"/>
<rosparam unless="$(arg baseline)" command="load" file="$(find hrc_simulator_moveit_config)/config/dirrt_planning.yaml" ns="go_to_location_server"/>-->

<rosparam command="load" file="$(find hrc_mosaic_test)/config/go_to_location_server.yaml"/>
<rosparam command="load" file="$(find hrc_mosaic_test)/config/locations_distribution.yaml"/>

<node pkg="go_to_location"  type="go_to_location_server" name="go_to_location_server" output="screen" launch-prefix="$(arg launch_prefix)" >
  <param if="$(eval pipeline == 'ompl')" name="ur5_on_guide/planning_plugin" value="ompl_interface/OMPLPlanner" />
  <param if="$(eval pipeline == 'human_aware')" name="ur5_on_guide/planning_plugin" value="ha_planner/DgacoPlannerManager" />
  <param if="$(eval pipeline == 'dirrt')" name="ur5_on_guide/planning_plugin" value="pathplan/DIRRTStar" />
  <param name="human_right_arm/planning_plugin" value="$(arg mao_planner)" />
  <remap from="/centroids" to="/centroid_detection_rs1/poses" />
  <remap from="/go_to_location_server/job/go_to" to="/robotiq_gripper"/>
</node>

<node pkg="go_to_location"  type="go_to_location_loader" name="go_to_location_loader" output="screen"/>
</launch>
